### PR TITLE
Makes welding plasma bars/statues/floors use flooder component

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
I originally wanted to do this in #60406 but didn't want to deal with the gameplay changes then.

Most plasma-derived items will now flood plasma, with movables also exploding. I don't really like how it's implemented code wise but at the same time I don't want to change the behavior significantly. Considered making this a prob but shooting your laser at a plasma bar with a 50% chance for it to work determined when spawned feels stupid.

~~Numbers are possibly a bit off ? I'll patch them up later.~~

Okay for mole numbers i looked into plasma bars and floors as a baseline. 
Use old bars as baseline: 10 per sheet, 50 per floor tile
Use old floors as baseline: 4 per sheet, 20 per floor tile
I chose 5 per sheet, 25 per tile. 250 a statue from old 50, but oh well the code taketh and the code giveth.

Equipment targeting check not implemented because I'm not sure we have any widely used right now (and lazy).

## Why It's Good For The Game
Makes things more consistent. Sort of a compromise solution to the outcomes I brought up in the linked PR.

## Changelog
:cl:
fix: Plasma material objects that didn't spawn plasma gas when ignited due to weird code reasons will now do just that.
fix: Fixed plasma material objects exploding when used with condenser.
fix: Fixed uranium manual doors and false walls not radiating.
fix: You can no longer duplicate rods with rough false walls.
code: Due to code structural changes the mole amount from welding plasma related objects may get buffed or nerfed. Mostly buffed.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/false_walls.dm [hunk 1]
```diff
@@ -315,8 +293,8 @@
 	icon = 'icons/turf/walls/iron_wall.dmi'
 	icon_state = "iron_wall-0"
 	base_icon_state = "iron_wall"
-	mineral = /obj/item/stack/rods
-	mineral_amount = 5
+	mineral = /obj/item/stack/sheet/iron
+	mineral_amount = 2
```
Comment (Fikou) [inline code/game/objects/structures/false_walls.dm:297]: jesse

Comment (Fikou) [reply code/game/objects/structures/false_walls.dm:297]: what the fuck are you doing

Comment (vincentiusvin) [reply code/game/objects/structures/false_walls.dm:297]: okay sorry

Comment (vincentiusvin) [reply code/game/objects/structures/false_walls.dm]: wait this DID fix the runtime, I'll find a better solution.

**Comment by vincentiusvin** [comment]
Having a globally available reference to initialized `obj/item/stack` entities would be better, I don't know if we already have one or not, do let me know if we do.

File: code/datums/components/combustible_flooder.dm [hunk 2]
```diff
@@ -33,24 +52,43 @@
...
-	if(isturf(parent))
-		var/turf/K = parent
-		K.ScrapeAway(1, CHANGETURF_INHERIT_AIR)
-	else
-		qdel(parent)
 
-/// Hotspot related flooding reaction.
+	if(delete && !QDELETED(parent))
+		qdel(parent) // For things with the explodable component like plasma mats this isn't necessary, but there's no harm. 
+	Destroy() // Dont want to stick around.
```
Comment (Mothblocks) [inline code/datums/components/combustible_flooder.dm:58]: Don't call Destroy() manually

```suggestion
	qdel(src)
```

File: code/datums/components/combustible_flooder.dm [hunk 3]
```diff
@@ -1,26 +1,45 @@
...
+	UnregisterSignal(parent, COMSIG_ATOM_TOOL_ACT(TOOL_WELDER))
+	UnregisterSignal(parent, COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER))
+	if(isturf(parent))
+		UnregisterSignal(parent, COMSIG_TURF_EXPOSE)
 
-/// Do the flooding.
-/datum/component/combustible_flooder/proc/flood(mob/user, temp_amount)
+/// Do the flooding. Trigger temperature is the temperature we will flood at if we dont have a temp set at the start. Trigger referring to whatever triggered it.
+/datum/component/combustible_flooder/proc/flood(mob/user, trigger_temperature)
+	var/delete = TRUE
```
Comment (Mothblocks) [inline code/datums/components/combustible_flooder.dm:33]: This only appears to delete the parent, you should clarify the variable name better.

File: code/datums/components/combustible_flooder.dm [hunk 4]
```diff
@@ -1,26 +1,45 @@
...
 	var/turf/open/flooded_turf = get_turf(parent)
-	flooded_turf.atmos_spawn_air("[gas_id]=[gas_amount];TEMP=[temp_amount]")
+
+	// We do this check early so closed turfs are still be able to flood.
+	if(isturf(parent)) // Walls and floors.
+		var/turf/parent_turf = parent
+		flooded_turf = parent_turf.ScrapeAway(1, CHANGETURF_INHERIT_AIR)
+		delete = FALSE
+
+	flooded_turf.atmos_spawn_air("[gas_id]=[gas_amount];TEMP=[temp_amount ? temp_amount : trigger_temperature]")
```
Comment (Mothblocks) [inline code/datums/components/combustible_flooder.dm:42]:
```suggestion
	flooded_turf.atmos_spawn_air("[gas_id]=[gas_amount];TEMP=[temp_amount || trigger_temperature]")
```

File: code/datums/components/combustible_flooder.dm [hunk 5]
```diff
@@ -1,26 +1,45 @@
...
-	src.temp_amount = temp_amount
+	src.gas_id = initialize_gas_id
+	src.gas_amount = initialize_gas_amount
+	src.temp_amount = initialize_temp_amount
 
 	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attackby_react)
 	RegisterSignal(parent, COMSIG_ATOM_FIRE_ACT, .proc/flame_react)
+	RegisterSignal(parent, COMSIG_ATOM_BULLET_ACT, .proc/projectile_react)
+	RegisterSignal(parent, COMSIG_ATOM_TOOL_ACT(TOOL_WELDER), .proc/welder_react)
+	RegisterSignal(parent, COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), .proc/welder_react)
```
Comment (Mothblocks) [inline code/datums/components/combustible_flooder.dm:18]: Why both? Should only be primary act IMO.

File: code/datums/components/explodable.dm [hunk 6]
```diff
@@ -22,6 +22,8 @@
 	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/explodable_attack)
 	RegisterSignal(parent, COMSIG_TRY_STORAGE_INSERT, .proc/explodable_insert_item)
 	RegisterSignal(parent, COMSIG_ATOM_EX_ACT, .proc/detonate)
+	RegisterSignal(parent, COMSIG_ATOM_TOOL_ACT(TOOL_WELDER), .proc/welder_react)
+	RegisterSignal(parent, COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), .proc/welder_react)
```
Comment (Mothblocks) [inline code/datums/components/explodable.dm:26]: Same here, should only be primary act.

File: code/datums/materials/basemats.dm [hunk 7]
```diff
@@ -137,11 +137,14 @@ Unless you know what you're doing, only use the first three numbers. They're in
 	. = ..()
 	if(ismovable(source))
 		source.AddElement(/datum/element/firestacker, amount=1)
-		source.AddComponent(/datum/component/explodable, 0, 0, amount / 2500, 0, amount / 1250)
+		// Ideally exploding plasma objects should delete themselves but we still have the flooder and SSexplosions to rely on deleting it asynchronously so it's not that bad.
+		source.AddComponent(/datum/component/explodable, 0, 0, amount / 2500, 0, amount / 1250, FALSE)
+	source.AddComponent(/datum/component/combustible_flooder, "plasma", amount*0.05, null)
```
Comment (Mothblocks) [inline code/datums/materials/basemats.dm:142]: The null here doesn't do anything

File: code/datums/materials/basemats.dm [hunk 8]
```diff
@@ -137,11 +137,14 @@ Unless you know what you're doing, only use the first three numbers. They're in
...
 
 /datum/material/plasma/on_removed(atom/source, amount, material_flags)
-	. = ..()
+	. = ..()
```
Comment (Mothblocks) [inline code/datums/materials/basemats.dm:145]:
```suggestion
	. = ..()
```

File: code/game/objects/structures/girders.dm [hunk 9]
```diff
@@ -59,14 +59,14 @@
 		if(istype(W, /obj/item/stack/rods))
 			var/obj/item/stack/rods/S = W
 			if(state == GIRDER_DISPLACED)
-				if(S.get_amount() < 2)
+				if(S.get_amount() < 5)
```
Comment (Mothblocks) [inline code/game/objects/structures/girders.dm:62]: Undocumented change? What is this for?

If this is for the duplicating rods change, this is an odd way to fix it, and feels like it should be a separate PR entirely.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I don't understand the other material changes, those feel like they should be a separate PR as they appear to have balance implications and are solving what appears to be a separate problem

File: code/game/objects/structures/girders.dm (hunk 9 above)
Comment (vincentiusvin) [reply code/game/objects/structures/girders.dm:62]: #35065 didnt do what it was meant to do. I was touching relevant code so i just slapped it in there.

Comment (vincentiusvin) [reply code/game/objects/structures/girders.dm:62]: Basing this off of #35065. It appears that it was simply deprecated so i just slapped it on there.

**Comment by vincentiusvin** [comment]
> I don't understand the other material changes, those feel like they should be a separate PR as they appear to have balance implications and are solving what appears to be a separate problem

By material changes do you mean just the rod one or the custom_materials one? Floydmats wasn't implemented for them yet and they will lose their flooding functionality if i didn't put it there.
If you meant the rod one I'm fine with atomizing it.

File: code/game/objects/structures/girders.dm (hunk 9 above)
Comment (Mothblocks) [reply code/game/objects/structures/girders.dm:62]: Oh, I see, this is for RODS, not metal. I don't care then

**Comment by Mothblocks** [comment]
Nevermind I didn't notice it was only rods
